x-pack/platform/plugins/shared/fleet/server/services/agent_policies/package_policies_to_agent_inputs.ts

```diff
@@ -125,8 +125,11 @@ export const getFullInputStreams = (
               const streamId = stream.id;
               const fullStream: FullAgentPolicyInputStream = {
                 id: streamId,
-                data_stream: stream.data_stream,
                 ...stream.compiled_stream,
+                data_stream: {
+                  ...stream.data_stream,
+                  ...stream.compiled_stream?.data_stream,
+                },
```

Comment:
Changing the order of these lines also changes the order of the generated policies, as seen in https://github.com/elastic/kibana/pull/214216/files#diff-cc408da9f76a79a5073e1f1d8cca2a983671a901abc14cc1bf7f5c603f8b5bb5R102

Is it a problem?

The resulting configuration is equivalent.

Reply:
Changing the order will change the priority of where the keys should come from, 


```
data_stream: {
                  ...stream.data_stream,
                  ...stream.compiled_stream?.data_stream, // will override any key defined in  stream.data_stream
                },
```

Reply:
I think it's what we want with that feature.
This make me think we probably need to make a change to the permissions too probably [here](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/fleet/server/services/agent_policies/package_policies_to_agent_permissions.ts#L161) otherwise agent will not have the correct permissions to write to the new type.

Reply:
Rewritten in https://github.com/elastic/kibana/pull/214216/commits/7eb26941aaf6e23f30651ad92e010d92eafd9dbc to avoid changing the order.